# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/AudioConduit.h
# Commit: e0a2e54e4de8
# Full Hash: e0a2e54e4de80813829cbcf9e4a5e4ff1285d451
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 9
# Description:
#   Bug 1654112 - Move WebRtcCallWrapper to dedicated files. r=ng
#   
#   This patch also renames it WebrtcCallWrapper, to better match neighboring
#   classes starting on "Webrtc".
#   
# ==============================================================================

diff -r 1965757ee924 -r e0a2e54e4de8 dom/media/webrtc/libwebrtcglue/AudioConduit.h
--- a/dom/media/webrtc/libwebrtcglue/AudioConduit.h	Thu Sep 02 13:22:03 2021 -0500
+++ b/dom/media/webrtc/libwebrtcglue/AudioConduit.h	Thu Apr 08 11:31:43 2021 +0200
@@ -45,7 +45,7 @@
    */
   void ReceivedRTCPPacket(const uint8_t* data, int len) override;
   Maybe<DOMHighResTimeStamp> LastRtcpReceived() const override;
-  DOMHighResTimeStamp GetNow() const override { return mCall->GetNow(); }
+  DOMHighResTimeStamp GetNow() const override;
 
   MediaConduitErrorCode StopTransmitting() override;
   MediaConduitErrorCode StartTransmitting() override;
@@ -143,24 +143,8 @@
 
   void DeleteStreams() override;
 
-  WebrtcAudioConduit(RefPtr<WebRtcCallWrapper> aCall,
-                     nsCOMPtr<nsISerialEventTarget> aStsThread)
-      : mTransportMonitor("WebrtcAudioConduit"),
-        mTransmitterTransport(nullptr),
-        mReceiverTransport(nullptr),
-        mCall(aCall),
-        mRecvStreamConfig(),
-        mRecvStream(nullptr),
-        mSendStreamConfig(
-            this)  // 'this' is stored but not  dereferenced in the constructor.
-        ,
-        mSendStream(nullptr),
-        mRecvSSRC(0),
-        mSendStreamRunning(false),
-        mRecvStreamRunning(false),
-        mDtmfEnabled(false),
-        mMutex("WebrtcAudioConduit::mMutex"),
-        mStsThread(aStsThread) {}
+  WebrtcAudioConduit(RefPtr<WebrtcCallWrapper> aCall,
+                     nsCOMPtr<nsISerialEventTarget> aStsThread);
 
   virtual ~WebrtcAudioConduit();
 
@@ -224,7 +208,7 @@
 
   // Const so can be accessed on any thread. Most methods are called on
   // main thread.
-  const RefPtr<WebRtcCallWrapper> mCall;
+  const RefPtr<WebrtcCallWrapper> mCall;
 
   // Accessed only on main thread.
   webrtc::AudioReceiveStream::Config mRecvStreamConfig;